ionc/ion_scanner.c

```diff
@@ -2161,15 +2161,19 @@ iERR _ion_scanner_read_possible_number(ION_SCANNER *scanner, int c, int sign, IO
 
     IONCHECK(_ion_scanner_read_char(scanner, &c));
 
+    if (c > 0 && c != 'x' && c != 'X' && c != 'b' && c != 'B' && !isdigit(c) && c != '_' && c != '-' && c != 'T'
+        && c != 'e' && c != 'E' && c != 'd' && c != 'D' && c != '.' && strchr(NUMERIC_STOP_CHARACTERS, c) == NULL ) {
+        FAILWITH(IERR_INVALID_SYNTAX);
+    }
```

Comment:
Is this branch exercised in tests? Is it necessary?

Reply:
I tested this branch again. it's unnecessary since we will check the last char at the end of the function anyway. I'll remove this redundant check. Thanks!